src/objects/avatar/AvatarLoader.ts

```diff
@@ -190,22 +192,26 @@ export class AvatarLoader implements IAvatarLoader {
       });
 
     directions.forEach((direction) => {
-      loadResources({
-        actions: new Set(actions),
-        direction,
-        look,
-        item,
-      });
+      directions.forEach(headDirection => {
+        loadResources({
+          actions: new Set(actions),
+          direction,
+          headDirection,
+          look,
+          item,
+        });
 
-      if (initial != null) {
-        preloadActions.forEach((action) => {
-          loadResources({
-            actions: new Set([action]),
-            direction,
-            look,
+        if (initial != null) {
+          preloadActions.forEach((action) => {
+            loadResources({
+              actions: new Set([action]),
+              direction,
+              headDirection,
+              look,
+            });
           });
-        });
-      }
+        }
+      })
```

Comment:
Is this needed? If we already loop through all the directions of the avatar, we should have gotten every head direction aswell if I'm not mistaken. So in theory we could do `headDirection: direction` and not loop `8*8` times.

Reply:
I believe this is required because it needs to cache each head direction for each body direction?

Reply:
I think one way to get aroundi t would be separate head and body rendering then stitch them together?

Reply:
In the `AvatarLoader` we don't cache based on combinations, but rather on the assets.
So I think when we load direction `2` for example, this will inevitably load the head assets for headDirection `2` aswell.

I don't think there is a case where a special combination of `direction` and `headDirection` would require other assets.
I could be wrong though.